### DISCUSSION TITLE
` --modules-dir` with `--node-linker hoisted`

### DISCUSSION THREAD

**Discussion by DreierF** [pnpm/pnpm#4921] [Q&A]
Hi,

Not really sure if this is bug or works as intended and I just don't understand the options correctly.
I'm trying to run `pnpm install --modules-dir copy_node_modules --node-linker hoisted`.

Using only `pnpm install --modules-dir copy_node_modules` correctly sets up a `copy_node_modules` folder with links to the pnpm store.
![image](https://user-images.githubusercontent.com/5631865/175538438-bf2bd7dd-d3aa-4bac-8786-5a8909816b00.png)

Running `pnpm install --node-linker hoisted` creates a `node_modules` folder as expected with actual folders instead of symlinks.
![image](https://user-images.githubusercontent.com/5631865/175539811-1587dd55-67be-4174-becc-1291694d0e75.png)

But combining both options results in

![image](https://user-images.githubusercontent.com/5631865/175538234-bd5d76ff-dc77-40d3-95e2-60a7706b0c28.png)

I would instead expect to get the actual folders inside `copy_node_modules`.

Am I misunderstanding one of both options or is this a bug?

Why?
We need a production dependencies only folder next to the regular node_modules, which also should contain dev dependencies, in a form that can be compressed into a zip archive.

Thanks in advance for any input

**Comment by MattSilvaa** [comment]
Did you ever figure this out?? I am trying to accomplish the same thing and noticed that this still occurs

**Reply by DreierF** [reply]
Unfortunately no. I worked around it by added a postprocessing step that moves the `node_modules` folder to the desired location after installing.

**Reply by MattSilvaa** [reply]
That's a bummer. I appreciate the response, I will probably end up doing something similar. Thank you!
